imsim/sag.py

```diff
@@ -0,0 +1,82 @@
+import galsim
+from galsim.config import ExtraOutputBuilder, RegisterExtraOutput, GetInputObj
+from galsim.config.input import ParseValue, GetAllParams
+import numpy as np
+import batoid
+from .telescope_loader import infer_optic_radii
+
+
+class SagBuilder(ExtraOutputBuilder):
+    """Build surface sag maps.
+
+    Required config inputs:
+        file_name: str
+            Name of file to write OPD images to.
+
+    Optional config inputs:
+        nx: int
+            Size of sag map images in pixels.  Default: 255.
+    """
+    def initialize(self, data, scratch, config, base, logger):
+        req = {'file_name': str}
+        opt = {'nx': int}
+        kwargs, safe = GetAllParams(config, base, req=req, opt=opt)
+        self.nx = kwargs.pop('nx', 255)
+        self.final_data = None
+
+    def finalize(self, config, base, main_data, logger):
+        telescope = GetInputObj(
+            'telescope',
+            config,
+            base,
+            'opd'
+        ).fiducial
+        self.final_data = []
+
+        xs = np.linspace(-1, 1, self.nx)
+
+        for name, optic in telescope.itemDict.items():
+            if not isinstance(optic, batoid.Interface):
+                continue
+            outer, inner = infer_optic_radii(optic)
+            xx = xs * outer
+            xx, yy = np.meshgrid(xx, xx)
+            rr = np.hypot(xx, yy)
+            ww = np.where((rr <= outer) & (rr >= inner))
+            out = np.nan * np.ones((self.nx, self.nx))
+            out[ww] = optic.surface.sag(xx[ww], yy[ww])
+
+            dx = (xs[1] - xs[0])*outer
+            dy = dx
+
+            x0, y0, z0 = optic.coordSys.origin
+            R = optic.coordSys.rot
+
+            sag_img = galsim.Image(np.array(out.data))
```

Comment:
Is there an appropriate pixel_scale to use here?  Maybe it never matters, but an Image typically has a scale, so worth considering if something makes sense in this case.  

Edit: I see dx, dy below, so maybe a jacobian WCS using dx,dy, which I suppose are allowed to be different.

Reply:
It's hard for me to imagine wanting different values for dx, dy, so I just made this a scale.